server/server.go

```diff
@@ -355,14 +355,54 @@ func (s *Server) CreateMetricsEndpoint() (*http.ServeMux, error) {
 	return mux, nil
 }
 
-// StopExitMonitor stops the exit monitor
-func (s *Server) StopExitMonitor() {
-	close(s.exitMonitorChan)
+// StopMonitors stops al the monitors
+func (s *Server) StopMonitors() {
+	close(s.monitorsChan)
 }
 
-// ExitMonitorCloseChan returns the close chan for the exit monitor
-func (s *Server) ExitMonitorCloseChan() chan struct{} {
-	return s.exitMonitorChan
+// MonitorsCloseChan returns the close chan for the exit monitor
+func (s *Server) MonitorsCloseChan() chan struct{} {
+	return s.monitorsChan
+}
+
+// StartHooksMonitor starts a goroutine to dinamycally add hooks at runtime
+func (s *Server) StartHooksMonitor() {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		logrus.Fatalf("Failed to create new watch: %v", err)
+	}
+	defer watcher.Close()
+
+	done := make(chan struct{})
+	go func() {
+		for {
+			select {
+			case event := <-watcher.Events:
+				logrus.Debugf("event: %v", event)
+				if event.Op&fsnotify.Remove == fsnotify.Remove {
+					logrus.Debugf("removing hook %s", event.Name)
+					s.ContainerServer.RemoveHook(filepath.Base(event.Name))
+				}
+				if event.Op&fsnotify.Create == fsnotify.Create || event.Op&fsnotify.Write == fsnotify.Write {
```

Comment:
If I modify a hook, say changes its path in the json, does this handle it correctly?

Reply:
yeah, cause fsnotify-go handles rename as "rename event + create event" so we should get an create one. Though, rename it's not something fully precise in the library :(